### PR TITLE
Add two macro rewrites for arrays

### PR DESCRIPTION
**Pull request by ajreynol**
These will be elaborated to simpler steps in future PRs.

### CODE REVIEW

File: include/cvc5/cvc5_proof_rule.h [hunk 1]
```diff
@@ -2498,6 +2498,32 @@ enum ENUM(ProofRewriteRule)
    * \endverbatim
    */
   EVALUE(ARRAYS_SELECT_CONST),
+
```
Comment (abdoo8080) [inline include/cvc5/cvc5_proof_rule.h:2501]:
```suggestion
```

File: include/cvc5/cvc5_proof_rule.h [hunk 2]
```diff
@@ -2498,6 +2498,32 @@ enum ENUM(ProofRewriteRule)
...
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Arrays -- Macro distinct arrays**
+   *
+   * .. math::
+   *   (A=B) = \bot
```
Comment (abdoo8080) [inline include/cvc5/cvc5_proof_rule.h:2507]:
```suggestion
   *   (A = B) = \bot
```

File: include/cvc5/cvc5_proof_rule.h [hunk 3]
```diff
@@ -2498,6 +2498,32 @@ enum ENUM(ProofRewriteRule)
...
+   *
+   * where :math:`A` and :math:`B` are distinct array values.
```
Comment (abdoo8080) [inline include/cvc5/cvc5_proof_rule.h:2509]: I think these descriptions are too vague...

File: src/theory/arrays/theory_arrays_rewriter.cpp [hunk 4]
```diff
@@ -76,6 +80,29 @@ Node TheoryArraysRewriter::rewriteViaRule(ProofRewriteRule id, const Node& n)
...
+      {
+        Assert(n[0].getType().isArray());
+        return d_nm->mkConst(false);
+      }
+    }
+    break;
+    case ProofRewriteRule::MACRO_ARRAYS_NORMALIZE_CONSTANT:
+    {
+      if (n.getType().isArray() && n[0].isConst() && n[1].isConst()
+          && n[2].isConst())
```
Comment (abdoo8080) [inline src/theory/arrays/theory_arrays_rewriter.cpp:96]: Is this a `store` operation? If so, how about just `n.isConst()`? If not, then I am worried about index out-of-bound error

File: include/cvc5/cvc5_proof_rule.h (hunk 3 above)
Comment (ajreynol) [reply include/cvc5/cvc5_proof_rule.h:2509]: Macro rules shouldn't require a precise explaination, but I can add that the internals are such that `Node::isConst` returns true for both?

File: src/theory/arrays/theory_arrays_rewriter.cpp (hunk 4 above)
Comment (ajreynol) [reply src/theory/arrays/theory_arrays_rewriter.cpp:96]: `n.isConst` is only true if the STORE is *canonical*.

I will test whether this is a store.

**Review by abdoo8080 — APPROVED** [review]

**Pull request merged**